jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/processor/DispatchingContentDirectory.java

```diff
@@ -197,99 +181,37 @@ public BrowseResult search(String containerId, String upnpSearchQuery, String fi
         UPnPSearchCriteria criteria = director.construct(offset, count, upnpSearchQuery);
 
         if (Artist.class == criteria.getAssignableClass()) {
-            return getArtistProcessor().toBrowseResult(searchService.search(criteria));
+            return artistProcessor.toBrowseResult(searchService.search(criteria));
         } else if (Album.class == criteria.getAssignableClass()) {
-            return getAlbumProcessor().toBrowseResult(searchService.search(criteria));
+            return albumProcessor.toBrowseResult(searchService.search(criteria));
         } else if (MediaFile.class == criteria.getAssignableClass()) {
-            return getMediaFileProcessor().toBrowseResult(searchService.search(criteria));
+            return mediaFileProcessor.toBrowseResult(searchService.search(criteria));
```

Comment:
Move toBrowseResult to Factory class.

Reply:
Hmm, at first glance it seems like this can be made common, but it might be better not to do it yet. The class that currently returns BrowseResult is clearly a subclass of  UPnPContentProcessor. DIDLContent, which is an intermediate object, may be transferred to Factory, but if BrowseResult is generated by Factory, the responsibilities between classes will be ambiguous.

Reply:
This is not fixed now.